### PR TITLE
docs: Added roadmap and a link to discussions.

### PR DESCRIPTION
**Pull request by iwoplaza**
https://github.com/user-attachments/assets/56f05646-9c9b-46e5-8da0-23c92fdf5298

### CODE REVIEW

File: apps/typegpu-docs/src/pages/index.astro [hunk 1]
```diff
@@ -150,6 +158,87 @@ const showcaseVideoResolution = [2048, 1200];
...
+            </RoadMapStep>
+            <RoadMapStep title="Pipelines">
+              <MilestonePipelinesBadge slot="visual" />
+              To ensure that shader stages match each-other, and pipeline constants
+              are filled with proper values, TypeGPU plans to provide typed pipelines.
+            </RoadMapStep>
+            <RoadMapStep title="Imperative code">
+              <MilestoneCodeBadge slot="visual" />
+              By giving developers the ability to write imperative shader code in
+              TypeScript, we can leverate the excellent TypeScript LSP to bring end-to-end
```
Comment (reczkok) [inline apps/typegpu-docs/src/pages/index.astro]:
```suggestion
              TypeScript, we can leverage the excellent TypeScript LSP to bring end-to-end
```

File: apps/typegpu-docs/src/pages/index.astro [hunk 2]
```diff
@@ -150,6 +158,87 @@ const showcaseVideoResolution = [2048, 1200];
...
+              typesafety to shader code like never before.
```
Comment (reczkok) [inline apps/typegpu-docs/src/pages/index.astro]: Shader code is type safe - the interaction with JS isn't. I would say we will bring type safety to applications utilizing gpu, not the shader code.

File: apps/typegpu-docs/src/components/roadmap/RoadMapStep.astro [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      class="hidden group-first:block absolute inset-0 bg-gradient-to-b from-[#5259EE] to-[#D6DDEF]">
+    </div>
+  </div>
+
+  <article>
+    <h3 class="flex items-center font-normal pb-3 md:pb-4">
+      {title}
+      {
+        releasedIn && (
+          <span class="ml-4 leading-normal font-sans font-light rounded-lg px-2 md:px-3 py-0.5 md:py-1 text-sm md:text-base lg:text-lg bg-[#5259EE] text-white">
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/components/roadmap/RoadMapStep.astro]:
```suggestion
          <span class="ml-4 leading-normal font-light rounded-lg px-2 md:px-3 py-0.5 md:py-1 text-sm md:text-base lg:text-lg bg-[#5259EE] text-white">
```

File: apps/typegpu-docs/src/pages/index.astro (hunk 2 above)
Comment (iwoplaza) [reply apps/typegpu-docs/src/pages/index.astro]: WDYM by interaction with JS?

File: apps/typegpu-docs/src/pages/index.astro [hunk 4]
```diff
@@ -150,6 +158,87 @@ const showcaseVideoResolution = [2048, 1200];
...
+      class="box-border flex flex-col items-center bg-white w-full my-32 py-8 md:p-16 lg:p-32 md:rounded-[5rem] rounded-[2.5rem]">
+      <section
+        class="box-border flex flex-col items-center justify-center px-4 lg:gap-20 gap-10 md:gap-[3.75rem] lg:max-w-[85rem] md:max-w-[53.5rem] max-w-[34.5rem]">
+        <div>
+          <h2 class="mb-4 font-normal text-center md:mb-8 lg:mb-8 center">
+            Roadmap
+          </h2>
+          <p class="text-center text-pretty max-w-[60rem]">
+            The road to end-to-end type safety on the GPU involves implementing
+            many interoperating primitives. Below is our roadmap of features
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/pages/index.astro]:
```suggestion
            several interoperating primitives. Below is our roadmap of features
```

File: apps/typegpu-docs/src/pages/index.astro (hunk 2 above)
Comment (reczkok) [reply apps/typegpu-docs/src/pages/index.astro]: The statement is correct, we will bring type safety to shader code - but `.wgsl` files are already type safe. WebGPU API isn't type safe - I would say we extend the type safety to the API land (while maintaining it inside the shader code)

File: apps/typegpu-docs/src/pages/index.astro [hunk 5]
```diff
@@ -150,6 +158,87 @@ const showcaseVideoResolution = [2048, 1200];
       </div>
     </section>
 
+    {/* Roadmap Section */}
+    <div
+      class="box-border flex flex-col items-center bg-white w-full my-32 py-8 md:p-16 lg:p-32 md:rounded-[5rem] rounded-[2.5rem]">
+      <section
+        class="box-border flex flex-col items-center justify-center px-4 lg:gap-20 gap-10 md:gap-[3.75rem] lg:max-w-[85rem] md:max-w-[53.5rem] max-w-[34.5rem]">
+        <div>
+          <h2 class="mb-4 font-normal text-center md:mb-8 lg:mb-8 center">
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/pages/index.astro]:
```suggestion
          <h2 class="mb-4 font-normal text-center md:mb-8 center">
```

File: apps/typegpu-docs/src/pages/index.astro [hunk 6]
```diff
@@ -150,6 +158,87 @@ const showcaseVideoResolution = [2048, 1200];
...
+            </RoadMapStep>
+            <RoadMapStep title="Functions">
+              <MilestoneFunctionsBadge slot="visual" />
+              By wrapping each function with a "typed shell", they can be moved across
+              different files/modules. Any resources (variables, buffers, etc.) in
+              the external scope can be accessed inside the function.
+            </RoadMapStep>
+            <RoadMapStep title="Pipelines">
+              <MilestonePipelinesBadge slot="visual" />
+              To ensure that shader stages match each-other, and pipeline constants
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/pages/index.astro]:
```suggestion
              To ensure that shader stages match each other, and pipeline constants
```

File: apps/typegpu-docs/src/pages/index.astro (hunk 1 above)
Comment (mhawryluk) [inline apps/typegpu-docs/src/pages/index.astro]:
```suggestion
              TypeScript, we can leverage the excellent TypeScript LSP to bring end-to-end
```

File: apps/typegpu-docs/src/pages/index.astro (hunk 2 above)
Comment (mhawryluk) [inline apps/typegpu-docs/src/pages/index.astro]:
```suggestion
              type safety to shader code like never before.
```

Comment (iwoplaza) [reply apps/typegpu-docs/src/pages/index.astro]: Aah, I see.

> I would say we extend the type safety to the API land (while maintaining it inside the shader code)

In my mind, all the previous milestones are responsible for maximizing type-safety in API land, and this one finally connects the two "type safeties" together. 🤔

Comment (iwoplaza) [reply apps/typegpu-docs/src/pages/index.astro]: > we can leverate the excellent TypeScript LSP to complete the type safety puzzle, bridging the gap between our API and type-safe shader code.

File: apps/typegpu-docs/src/pages/index.astro [hunk 7]
```diff
@@ -150,6 +158,87 @@ const showcaseVideoResolution = [2048, 1200];
...
+          <p class="text-center text-pretty max-w-[50rem]">
+            Take active part in the development by joining our discussions.
+          </p>
+          <div class="flex flex-col items-center sm:flex-row gap-3">
+            <a
+              class="inline-flex justify-between items-center rounded-[6.25rem] gap-3 box-border sm:w-auto text-black/75 py-4 px-6 md:py-5 md:px-7 lg:py-6 lg:px-8 transition-all shadow hover:shadow-md shadow-tameplum-20 bg-white hover:text-black text-center text-base md:text-xl lg:text-2xl no-underline"
+              href="https://github.com/software-mansion/TypeGPU/discussions"
+              target="_blank"
+              rel="noreferrer"
+              ><GitHubIcon class="w-7 h-auto" /> GitHub Discussions</a
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/pages/index.astro]: why not use the github and discord icons we already have in the project?

**Review by mhawryluk — COMMENTED** [review]
Zrzut ekranu 2024-10-22 o 15 59 15

Zrzut ekranu 2024-10-22 o 16 00 09

imo the spacing should be adjusted for the small and medium breakpoints. there is too much space before and after the roadmap section and not enough padding inside it above the header and below the buttons

File: apps/typegpu-docs/src/pages/index.astro (hunk 2 above)
Comment (iwoplaza) [reply apps/typegpu-docs/src/pages/index.astro]: @reczkok Thoughts? 👀

**Review by mhawryluk — APPROVED** [review]
Stunning 🚀

**Pull request merged**